src/core/database.py

```diff
@@ -1,16 +1,38 @@
+import numpy as np
 import open3d as o3d
 
 from dataclasses import dataclass
+from functools import cached_property
 from nptyping import Float, NDArray, Shape, UInt8
-from src.core.image import Image
-from src.core.point_cloud import PointCloud
+from src.core.cache import memory
+from src.core.image_provider import ImageProvider
```

Comment:
Looks like in that case all providers should go out of core, maybe move to `loaders`/`readers` (btw, `loaders` is more widely used name)  since it is connected to format?